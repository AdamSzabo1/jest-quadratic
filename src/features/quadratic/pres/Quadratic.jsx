
import CoeInput from '../coeInput/CoeInput';
import QuadOutput from '../quadOutput/QuadOutput';
import * as style from './quadratic.module.css';

const Quadratic = ({ a, b, c, D, x1, x2 }) => {

    return (
        <div className={`${style.quadratic}`}>
            <h1 data-testid="quadratic-heading">Quadratic Equation</h1>

            <div className={`${style['io-area']}`}>

                <CoeInput
                    defA={a}
                    defB={b}
                    defC={c}
                />
                <QuadOutput
                    a={a}
                    b={b}
                    c={c}
                    D={D}
                    x1={x1}
                    x2={x2}
                />

            </div>
        </div>
    );
}

export default Quadratic;
