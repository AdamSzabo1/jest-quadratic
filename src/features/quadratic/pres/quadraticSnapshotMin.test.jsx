import React from "react";
import { render } from "@testing-library/react";
import '@testing-library/jest-dom';

jest.mock('katex', () => ({
    ...jest.requireActual('katex'),
    render: jest.fn(() => {})
}));

import Quadratic from "./Quadratic";

describe('Quadratic snapshot min', () => {

    test('should match previous snapshot without KaTeX', () => {

        const { asFragment } = render(
            <Quadratic
                a={1}
                b={8}
                c={-48}
                D={256}
                x1={4}
                x2={-12}
            />
        );

        expect( asFragment() ).toMatchSnapshot();
    });

});
