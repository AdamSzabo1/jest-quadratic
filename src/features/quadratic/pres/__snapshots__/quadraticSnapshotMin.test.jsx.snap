// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quadratic snapshot min should match previous snapshot without KaTeX 1`] = `
<DocumentFragment>
  <div
    class="undefined"
  >
    <h1
      data-testid="quadratic-heading"
    >
      Quadratic Equation
    </h1>
    <div
      class="undefined"
    >
      <form>
        <h2>
          Input
        </h2>
        <div>
          <label
            for="coe-a"
          >
            a: 
          </label>
          <input
            id="coe-a"
            name=""
            type="number"
            value="1"
          />
        </div>
        <div>
          <label
            for="coe-b"
          >
            b: 
          </label>
          <input
            id="coe-b"
            name=""
            type="number"
            value="8"
          />
        </div>
        <div>
          <label
            for="coe-c"
          >
            c: 
          </label>
          <input
            id="coe-c"
            name=""
            type="number"
            value="-48"
          />
        </div>
        <button
          type="submit"
        >
          Solve
        </button>
      </form>
      <div>
        <h2>
          Solution
        </h2>
        <span />
        <span />
        <span />
        <span>
          x1 = 4
        </span>
        <span>
          x2 = -12
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
